src/nri/sampler_test.go

```diff
@@ -111,3 +122,80 @@ func TestECSLabelRename(t *testing.T) {
 		}
 	}
 }
+
+func TestExitedContainerTTL_Expired(t *testing.T) {
+	mocker := &mocker{}
+	mocker.On("ContainerList", mock.Anything, mock.Anything).Return([]types.Container{
+		{
+			ID:      "containerid",
+			Names:   []string{"Container 1"},
+			Image:   "my_image",
+			ImageID: "my_image_id",
+		},
+	}, nil)
+	mocker.On("ContainerInspect", mock.Anything, mock.Anything).Return(types.ContainerJSON{
+		ContainerJSONBase: &types.ContainerJSONBase{
+			State: &types.ContainerState{
+				Status:     "exited",
+				FinishedAt: time.Now().Add(-2 * time.Hour).Format(time.RFC3339Nano),
+			},
+		},
+	}, nil)
+
+	mStore := &mockStorer{}
+	mStore.On("Save").Return(nil)
+
+	sampler := ContainerSampler{
+		metrics: biz.NewProcessor(mStore, nil, mocker, 30*time.Minute),
+		docker:  mocker,
+		store:   mStore,
+	}
+
+	i, err := integration.New("test", "test-version")
+	assert.NoError(t, err)
+
+	err = sampler.SampleAll(context.Background(), i)
+	assert.Empty(t, i.Entities)
+}
+
+func TestSampleAll(t *testing.T) {
+	mocker := &mocker{}
+	mocker.On("ContainerList", mock.Anything, mock.Anything).Return([]types.Container{
+		{
+			ID:      "containerid",
+			Names:   []string{"Container 1"},
+			Image:   "my_image",
+			ImageID: "my_image_id",
+		},
+	}, nil)
+	mocker.On("ContainerInspect", mock.Anything, mock.Anything).Return(types.ContainerJSON{
+		ContainerJSONBase: &types.ContainerJSONBase{
+			State: &types.ContainerState{
+				Status:     "exited",
+				FinishedAt: time.Now().Add(15 * time.Minute).Format(time.RFC3339Nano),
```

Comment:
This would cause the `FinishedAt` timestamp to be in the future. While it should work just fine for this case, I don't think it's the case we want to test
```suggestion
				FinishedAt: time.Now().Add(-15 * time.Minute).Format(time.RFC3339Nano),
```